src/api/private/auth/auth.controller.ts

```diff
@@ -70,14 +70,20 @@ export class AuthController {
     @Body() changePasswordDto: UpdatePasswordDto,
   ): Promise<void> {
     try {
+      await this.identityService.loginWithLocalIdentity(
+        user,
+        changePasswordDto.currentPassword,
+      );
       await this.identityService.updateLocalPassword(
         user,
         changePasswordDto.newPassword,
       );
       return;
     } catch (e) {
       if (e instanceof NotInDBError) {
-        throw new NotFoundException(e.message);
+        throw new UnauthorizedException(
+          'Verifying your identity with the current password did not work.',
```

Comment:
Isn't it bad practice to tell people that their problem lies with the password, even if we literally do that every time here even when the user does not exist?

Reply:
No, this is the password change not the login.
The user must exist because it is retrieved from the session, that is checked by the guard.
The user also does not enter their username but only their current password and their new password. Therefore only the current password could be wrong.

Reply:
One could argue however whether it is appropriate that the `identityService.loginWithLocalIdentity` method throws a `NotInDBError` in case of invalid username or password. A `InvalidCredentialsError` seems more meaningful to me.

Reply:
I'd suggest adding an issue for this?

Reply:
#1936